bpf/tests/tc_egressgw_redirect.c

```diff
@@ -366,7 +366,7 @@ int egressgw_skip_no_gateway_redirect_check(const struct __ctx_buff *ctx)
 		test_fatal("status code out of bounds");
 
 	status_code = data;
-	assert(*status_code == CTX_ACT_OK);
+	assert(*status_code == CTX_ACT_DROP);
```

Comment:
I should probably check the reason here but I am not sure how. Maybe similar code than this https://github.com/cilium/cilium/blob/master/bpf/tests/tc_lxc_policy_drop.c#L134 could work?

Reply:
Yup, the drop reason is passed to userspace through a perf event map, but the metrics map should get updated as well so checking that should work :+1: